test/controllers/teacher_controller_test.exs

```diff
@@ -66,4 +101,44 @@ defmodule CoursePlanner.TeacherControllerTest do
     conn = get conn, teacher_path(conn, :new)
     assert html_response(conn, 200) =~ "New teacher"
   end
+
+  test "gets empty list when teacher has no course assigned" do
+    teacher = Repo.insert!(%{@user | role: "Teacher"})
```

Comment:
Can you insert through `Teachers.new/2` with whatever string as second parameter?